### PR TITLE
feat: Added warning message when multiple locations found for an item

### PR DESCRIPTION
**Pull request by balloonpopper**
Prints a warning if there is more than one ESCLocation (position2d) as a child of an ESCITem

### CODE REVIEW

**Comment by balloonpopper** [comment]
Fixes https://github.com/godot-escoria/escoria-issues/issues/202

File: addons/escoria-core/game/core-scripts/esc_item.gd [hunk 1]
```diff
@@ -379,14 +379,21 @@ func get_animation_player() -> Node:
...
 	for c in get_children():
 		if c is Position2D:
+			if interact_position != null:
+				multiple_positions_found = true
 			interact_position = c.global_position
 
 	if interact_position == null and collision != null:
 		interact_position = collision.global_position
 
+	if multiple_positions_found == true:
```
Comment (BHSDuncan) [inline addons/escoria-core/game/core-scripts/esc_item.gd:393]:
```suggestion
	if multiple_positions_found:
```

File: addons/escoria-core/game/core-scripts/esc_item.gd [hunk 2]
```diff
@@ -379,14 +379,21 @@ func get_animation_player() -> Node:
...
+		escoria.logger.warning("Multiple ESClocations found to walk to for object " +  global_id + \
```
Comment (BHSDuncan) [inline addons/escoria-core/game/core-scripts/esc_item.gd:394]: Prefer string interpolation to concatenation; i.e. make use of the `%` operator.

**Review by BHSDuncan — CHANGES_REQUESTED** [review]

File: addons/escoria-core/game/core-scripts/esc_item.gd (hunk 2 above)
Comment (balloonpopper) [reply addons/escoria-core/game/core-scripts/esc_item.gd:394]: It didn't work when I tried it as string interpolation.

Comment (balloonpopper) [reply addons/escoria-core/game/core-scripts/esc_item.gd:394]: It didn't work when I tried it as string interpolation.

Comment (BHSDuncan) [reply addons/escoria-core/game/core-scripts/esc_item.gd:394]: Not sure why it wouldn't...what was the exact line? I know that you can for sure use string interpolation like:

```
var s = "here"
escoria.logger.warning("Using interpolation %s" % s)
```

Comment (balloonpopper) [reply addons/escoria-core/game/core-scripts/esc_item.gd]: My mistake, had a comma instead of percent when I coded it the first time. Oops 🤦

**Comment by balloonpopper** [comment]
Added a couple of optimisations in - the function was getting called 4 times for the one walk operation. The optimisations mean it gets called once.

File: addons/escoria-core/game/core-scripts/esc/esc_action_manager.gd [hunk 3]
```diff
@@ -620,10 +616,14 @@ func _walk_towards_object(
 	var destination_position: Vector2
 	var dont_interact: bool = false
 
+	if obj == null || obj.node == null:
+		escoria.logger.report_errors("esc_action_manager.gd:_walk_towards_object",
+			["obj or obj.node not populated."])
```
Comment (BHSDuncan) [inline addons/escoria-core/game/core-scripts/esc/esc_action_manager.gd:621]:
```suggestion
	if obj == null or obj.node == null:
		escoria.logger.report_errors("esc_action_manager.gd:_walk_towards_object",
			["obj or obj.node not populated."])

```

**Review by BHSDuncan — CHANGES_REQUESTED** [review]

**Review by BHSDuncan — APPROVED** [review]

**Review by StraToN — APPROVED** [review]

**Comment by StraToN** [comment]
Thanks a lot! 🎈

**Pull request merged**